optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -180,7 +190,332 @@
      * @return a stream that matches every combination of A and B for which all the {@link BiJoiner joiners} are true
      */
     default <B> BiConstraintStream<A, B> join(Class<B> otherClass, BiJoiner<A, B>... joiners) {
-        return join(otherClass, AbstractBiJoiner.merge(joiners));
```

Comment:
There is a `join(Class<B>, joiners)` and a `join(Uni<B>, joiners)`.
Do both support filtering() joiners now? Isn't this new code below here duplicated between them?

It feels to me like we want to extract these new lines below into a separate non-public method if possible.

Reply:
This is an interface and Java 8, so there are no non-public methods.
Yes, all the `join()` methods support `filtering()`. In fact, `filtering()` is converted to `join().filter()`, so the JoinStream never even gets to see that particular joiner.

Reply:
Good that the JoinStream never sees that partical joiner.

I am just wondering - if both join(Class) and join(Stream) handle the FilteringJoiner-extraction, do they duplicate the code below? If so, can we de-duplicate it by letting this one delegate to the Stream one?